File: LibraryApp/app/src/main/java/com/example/libraryapp/Report.java
package com.example.libraryapp;

import java.util.Date;

public class Report {
    private String userId;
    private String userName;
    private String issue;
    private Date timestamp;

    public Report() {} // Required for Firebase

    public Report(String userId, String userName, String issue, Date timestamp) {
        this.userId = userId;
        this.userName = userName;
        this.issue = issue;
        this.timestamp = timestamp;
    }

    public String getUserId() { return userId; }
    public String getUserName() { return userName; }
    public String getIssue() { return issue; }
    public Date getTimestamp() { return timestamp; }
}

File: src/main/java/com/example/libraryapp20/Report.kt
package com.example.libraryapp20

import java.util.Date


class Report {
    var userId: String? = null
        private set
    var userName: String? = null
        private set
    var issue: String? = null
        private set
    var timestamp: Date? = null
        private set

    constructor() // Required for Firebase

    constructor(userId: String?, userName: String?, issue: String?, timestamp: Date?) {
        this.userId = userId
        this.userName = userName
        this.issue = issue
        this.timestamp = timestamp
    }
}